src/ZxSpectrumHidJoystick.h

```diff
@@ -7,15 +7,22 @@ class ZxSpectrumHidJoystick : public ZxSpectrumJoystick {
 private:
   uint32_t _updated1;
   uint32_t _updated2;
+  uint8_t _joy1_old;  
```

Comment:
Is this still used?

Reply:
Yes, it hold previous state for joystick contact, it is nessesary for detaction of press button  - the edge of signal.

Reply:
...the previous state appears to be held in PicoWinHidKeyboard:

```
int PicoWinHidKeyboard::processJoystick(uint8_t value) {
  int r = 0;

  if (value& JOYSTICK_WANT) {
    if (value&JOYSTICK_RIGHT)  if (!(old_value&JOYSTICK_RIGHT))  keyPressed(0, 0, ASCII_RIGHT);
    if (value&JOYSTICK_LEFT)  if (!(old_value&JOYSTICK_LEFT))  keyPressed(0, 0, ASCII_LEFT);
    if (value&JOYSTICK_UP)  if (!(old_value&JOYSTICK_UP))  keyPressed(0, 0, ASCII_UP);
    if (value&JOYSTICK_DOWN)  if (!(old_value&JOYSTICK_DOWN))  keyPressed(0, 0, ASCII_DOWN);
    if (value&JOYSTICK_BT0)  if (!(old_value&JOYSTICK_BT0))  r = 1; //Fire to escape from Menu   
  }
  old_value=value;

  return r;
}
```

I can see _joy1_old assigned but never read.